app/assets/javascripts/heatmap.js

```diff
@@ -1,91 +1,62 @@
+    if (weekdayRow % 2 === 0) $day.addClass("d-none");
+    $day.css("grid-area", weekdayRow + " / 1");
   }
 
-  /** @type {{date: string; max_id: number; total_changes: number}[]} */
-  const heatmapData = heatmapElement.dataset.heatmap ? JSON.parse(heatmapElement.dataset.heatmap) : [];
-  const displayName = heatmapElement.dataset.displayName;
-  const colorScheme = document.documentElement.getAttribute("data-bs-theme") ?? "auto";
-  const rangeColorsDark = ["#14432a", "#4dd05a"];
-  const rangeColorsLight = ["#4dd05a", "#14432a"];
-  const startDate = new Date(Date.now() - (365 * 24 * 60 * 60 * 1000));
-
-  const mediaQuery = window.matchMedia("(prefers-color-scheme: dark)");
-
-  let cal = new CalHeatmap();
-  let currentTheme = getTheme();
-
-  function renderHeatmap() {
-    cal.destroy();
-    cal = new CalHeatmap();
-
-    cal.paint({
-      itemSelector: "#cal-heatmap",
-      theme: currentTheme,
-      domain: {
-        type: "month",
-        gutter: 4,
-        label: {
-          text: (timestamp) => new Date(timestamp).toLocaleString(OSM.i18n.locale, { timeZone: "UTC", month: "short" }),
-          position: "top",
-          textAlign: "middle"
-        },
-        dynamicDimension: true
-      },
-      subDomain: {
-        type: "ghDay",
-        radius: 2,
-        width: 11,
-        height: 11,
-        gutter: 4
-      },
-      date: {
-        start: startDate
-      },
-      range: 13,
-      data: {
-        source: heatmapData,
-        type: "json",
-        x: "date",
-        y: "total_changes"
-      },
-      scale: {
-        color: {
-          type: "sqrt",
-          range: currentTheme === "dark" ? rangeColorsDark : rangeColorsLight,
-          domain: [0, Math.max(0, ...heatmapData.map(d => d.total_changes))]
-        }
+  for (const day of heatmap.find("[data-date]")) {
+    const $day = $(day);
+    const date = new Date($day.data("date"));
+    if (date.getUTCDay() === weekInfo.firstDay) {
+      weekColumn++;
+      const currentMonth = getMonthOfThisWeek(date);
+      if (previousMonth !== currentMonth) {
+        heatmap.find(`[data-month="${previousMonth}"]`).css("grid-column-end", weekColumn);
+        heatmap.find(`[data-month="${currentMonth}"]`).css("grid-column-start", weekColumn);
+        previousMonth = currentMonth;
       }
-    }, [
-      [Tooltip, {
-        text: (date, value) => getTooltipText(date, value)
-      }]
-    ]);
-
-    cal.on("mouseover", (event, timestamp, value) => {
-      if (!displayName || !value) return;
-      if (event.target.parentElement.nodeName === "a") return;
+    }
+    if (weekColumn === 1) {
+      $day.addClass("d-none");
+      continue;
+    }
+    const count = $day.data("count") ?? 0;
+    const tooltipText = getTooltipText(date, count);
+    $day
+      .css("grid-area", getWeekdayRow(date.getUTCDay()) + " / " + weekColumn)
+      .attr("aria-label", tooltipText)
+      .tooltip({
+        title: tooltipText,
```

Comment:
Currently the tooltips are prone to line wrapping which doesn't look very good so I'd suggest adding a custom class here like:

```js
customClass: "heatmap-tooltip"
```

and then a CSS rule to allow wider tooltips:

```css
.heatmap-tooltip {
  --bs-tooltip-max-width: 300px;
}
```